stan/math/prim/core/init_threadpool_tbb.hpp

```diff
@@ -88,12 +88,20 @@ inline int get_num_threads() {
  * The function returns a reference to the static
  * tbb::global_control instance.
  *
+ * @param n_threads The maximum number of threads available to the tbb. If not
+ *  set will search for the environment variable `STAN_NUM_THREADS`. A value of
+ *  will assume all detectable threads are available for the process.
  * @return reference to the static tbb::global_control
  * @throws std::runtime_error if the value of STAN_NUM_THREADS env. variable
  * is invalid
  */
-inline tbb::task_arena& init_threadpool_tbb() {
-  int tbb_max_threads = internal::get_num_threads();
+inline tbb::task_arena& init_threadpool_tbb(size_t n_threads = 0) {
+  int tbb_max_threads;
+  if (n_threads == 0) {
+    tbb_max_threads = internal::get_num_threads();
+  } else {
+    tbb_max_threads = n_threads;
+  }
```

Comment:
Need to check if `n_threads` is a valid integer greater than zero and smaller than or equal to the maximum number of threads supported by the hardware (`0 <= n_threads <= tbb::this_task_arena::max_concurrency()`).

Reply:
I updated this to account for the above and to allow for n_threads = -1 to mean `std::thread::hardware_concurrency()`